src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -347,10 +347,19 @@ export default class DeployFromSettingsController {
    */
   getContainerImageVersion_() {
     /** @type {number} */
-    let index = (this.containerImage || '').lastIndexOf(':');
+    let index = (this.containerImage || '').lastIndexOf('/');
```

Comment:
Second test actually shows that even if version is provided in the link it will be parsed as empty string, so potential version information is lost. Moreover checking only for `:` or `/` or some other char will be measured only for certains cases and not fully generic.

This library is a good idea and I'd go with that.